tests/integration/standard/test_policies.py

```diff
@@ -102,3 +111,26 @@ def test_speculative_execution(self):
         # Test timeout with spec_ex
         with self.assertRaises(OperationTimedOut):
             result = self.session.execute(statement, execution_profile='spec_ep_rr', timeout=.5)
+
+    #TODO redo this tests with Scassandra
+    def test_speculative_and_timeout(self):
+        """
+        Test to ensure the timeout is honored when using speculative execution
+        @since 3.10
+        @jira_ticket PYTHON-750
+        @expected_result speculative retries be schedule every fixed period, during the maximum
+        period of the timeout.
+
+        @test_category metadata
+        """
+        # We mock this so no messages are sent, otherwise a reponse might arrive
+        # and we would not know how many hosts we queried
+        with patch.object(Connection, "send_msg") as mocked_send_msg:
+            mocked_send_msg.return_value = 100
+
+            statement = SimpleStatement("INSERT INTO test3rf.test (k, v) VALUES (0, 1);", is_idempotent=True)
+            response_future = self.session.execute_async(statement, execution_profile='spec_ep_brr_lim', timeout=2.2)
+            response_future._event.wait()
```

Comment:
The problem with this is that an OperationTimedOut is arisen and later it's harder to count the number of queried hosts